pkg/registries/adapters/openshift_adapter.go

```diff
@@ -0,0 +1,112 @@
+package adapters
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	b64 "encoding/base64"
+	logging "github.com/op/go-logging"
+	"github.com/openshift/ansible-service-broker/pkg/apb"
+)
+
+const openShiftName = "OpenShift"
+const openShiftAuthURL = "https://sso.redhat.com/auth/realms/rhc4tp/protocol/docker-v2/auth?service=docker-registry"
+const openShiftManifestURL = "https://registry.connect.redhat.com/v2/%v/manifests/latest"
+
+// OpenShiftAdapter - Docker Hub Adapter
+type OpenShiftAdapter struct {
+	Config Configuration
+	Log    *logging.Logger
+}
+
+// OpenShiftImage - Image from a OpenShift registry.
+type OpenShiftImage struct {
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
+}
+
+// OpenShiftImageResponse - Image response for OpenShift.
+type OpenShiftImageResponse struct {
+	Count   int               `json:"count"`
+	Results []*OpenShiftImage `json:"results"`
+	Next    string            `json:"next"`
+}
+
+// RegistryName - Retrieve the registry name
+func (r OpenShiftAdapter) RegistryName() string {
+	return openShiftName
+}
+
+// GetImageNames - retrieve the images
+func (r OpenShiftAdapter) GetImageNames() ([]string, error) {
+	r.Log.Debug("OpenShiftAdapter::GetImages")
+	r.Log.Debug("BundleSpecLabel: %s", BundleSpecLabel)
+
+	images := r.Config.Images
```

Comment:
Config.Images is used because there is no way to search an openshift registry. This allows us to source images that are in a registry in the config.